Core/StitchCoreSDK/Sources/StitchCoreSDK/Internal/Net/RawSSEStream.swift

```diff
@@ -0,0 +1,148 @@
+import Foundation
+
+private let newlineChar = [UInt8]("\n".utf8)[0]
+
+public enum SSEStreamState {
+    case open, closed
+}
+
+open class SSEStreamDelegate {
+    public init() {}
+    
+    open func on(newEvent event: RawSSE) {
+
+    }
+
+    open func on(stateChangedFor state: SSEStreamState) {
+
+    }
+
+    open func on(error: Error) {
+
+    }
+}
+
+open class RawSSEStream {
+    open var state: SSEStreamState = .closed {
+        didSet {
+            delegate?.on(stateChangedFor: state)
+        }
+    }
+    public weak var delegate: SSEStreamDelegate? = nil
+    public var dataBuffer = Data()
+
+    private var stringBuffer: String = ""
+    private var eventName: String = ""
```

Comment:
[nit] `eventNameBuffer`. Otherwise it makes this class seem like just an event class